aten/src/ATen/native/cpu/BinaryOpsKernel.cpp

```diff
@@ -103,6 +103,18 @@ void logical_xor_kernel(TensorIterator& iter) {
     });
 }
 
+void lt_kernel(TensorIterator& iter) {
+  AT_DISPATCH_ALL_TYPES_AND2(kBool, kBFloat16, iter.dtype(), "lt_cpu_result_", [&] {
```

Comment:
Using the dispatch macro twice results in a lot of code. Each macro is a big switch statement, and having two macros like this means there are `n^2` amount of code, where `n` is the number of types.

Could we just have `return_t = bool` and avoid the outer dispatch? There should be an implicit conversion somewhere (I am inferring this from the code below) from bool to any `scalar_t`.

Same thing applies to the CUDA kernel.

Reply:
Fixed